Enterprise-CMCS/managed-care-review
# MCR-4612: Approval Modal Working with API

## Summary

Note: should be merged after https://github.com/Enterprise-CMCS/managed-care-review/pull/2925

As a CMS user,
I want to mark a submission as approved
so that I can more easily keep track of which submissions in MC-Review are approved and which are still active.

AC: 

- If I am viewing the approval confirmation modal and I select "approve submission"
then the modal closes and I am still on the submission summary page but the "approve submission" button no longer appears on the submission summary
- If I select "approve submission"
- Then the reviewStatus for the submission in the database is set to "Approved" (the separate status field should still be set to "Submitted")
- If I click "cancel"
Then the modal closes and the "approve submission" button still appears and the status of the submission in the database is still "Submitted"

Other notes
- Did get confirmation from Kelsey that it's okay the optional note box is standard size and there's not character limit
- This PR also implements https://jiraent.cms.gov/browse/MCR-4592 
- Given that the "UnlockSubmitModal" now handles approval too, I changed the name to "ActionModal"

#### Related issues

https://jiraent.cms.gov/browse/MCR-4612

#### Screenshots

<img width="861" alt="Screenshot 2024-11-15 at 11 14 50 AM" src="https://github.com/user-attachments/assets/7d43a368-2c01-4088-9008-336e97149675">

<img width="519" alt="Screenshot 2024-11-15 at 11 19 32 AM" src="https://github.com/user-attachments/assets/a5f017a4-2ec8-4444-bc94-b2690fc1a423">


#### Test cases covered

Modal displays expected heading 

## QA guidance

Click 'Approve Submission' and check that the reviewStatus changes to 'APPROVED' and the reviewStatusActions field has a new entry. The updatedReason note should also appear if that optional field was used
